### ISSUE TITLE
Use userdata for storing what gadgets a player has unlocked

### ISSUE THREAD

**Issue by RoboMWM** [OnlyNoobDied/GadgetsMenu#633]
Unless I'm missing a config option, I don't see any reason why the plugin must run a command to give an individual player permission for a gadget when they "unlock" it - would make a lot more sense to just store what gadgets they have unlocked in the already-existing userdata and keep permissions cleaner.

If anything, this design is an abuse of the permission system. Permissions define what a player does or does not have access to, and is statically defined. E.g., permissions could be used to determine what things a player can unlock (e.g. could use ranks to allow for more unlockables) or use (e.g. prevent usage of some gadgets in some worlds) as some other people have requested. Individual permissions shouldn't be dynamically added; is there any other plugin that sets permissions to store data?

**Comment by OnlyNoobDied** [comment]
Actually I still thinking this problem. What if I add custom loots into Mystery Boxes, how can it work?

**Comment by RoboMWM** [comment]
Sorry, can you rephrase what you're asking? I don't understand the question.

If you're asking about user-defined custom loot, I'd assume they would define the name of the loot in the plugin, so I don't see how this is an issue...?

**Comment by OnlyNoobDied** [comment]
I planning to add this feature for months. However, there're several problems when I switch the permission system.

**Comment by RoboMWM** [comment]
Can you explain what you mean by custom loots?

**Comment by RoboMWM** [comment]
Still not sure what you mean by custom loots. 

If the custom loots would be defined in the plugin/config file, then the name of the custom loot could be stored and used to determine if they unlocked it or not.

**Comment by OnlyNoobDied** [comment]
This feature will be added in the next update.
